src/libsystemd/sd-id128/id128-util.c

```diff
@@ -13,6 +13,23 @@
 #include "string-util.h"
 #include "sync-util.h"
 
+int id128_from_string_not_null(const char *s, sd_id128_t *ret) {
+        sd_id128_t t;
+        int r;
+
+        assert(ret);
+
```

Comment:
maybe treat s == NULL also as reason to return ENXIO?

Reply:
There's ASSERT_PTR. This is for internal code, so I think assert is appropriate.